### PR TITLE
disttask: replace spool with waitgroup

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #49008

Problem Summary:

### What changed and how does it work?
- we already handle how many schedulers we can start, we don't need spool which introduces meaningless error handling
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/disttask/framework/taskexecutor/manager.go [hunk 1]
```diff
@@ -270,18 +240,11 @@ func (m *Manager) handleExecutableTasks(tasks []*proto.Task) {
 		m.addHandlingTask(task.ID)
 		m.slotManager.alloc(task)
 		t := task
-		err = m.executorPool.Run(func() {
+		m.executorWG.Go(func() {
```
Comment (zimulala) [inline pkg/disttask/framework/taskexecutor/manager.go:243]: Could we rename `Go` to `RunWithLog` or other names which is unify `WaitGroupWrapper`'s functions?

I don't think this function is used anywhere else, so I just mentioned it in this PR.

Comment (D3Hunter) [reply pkg/disttask/framework/taskexecutor/manager.go]: https://github.com/pingcap/tidb/pull/50664/commits/e7228e463a86d03ff5399e35315897b9b1cd1d3b

**Review by zimulala — APPROVED** [review]

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**